webapp/src/components/admin_settings/job_table/job_table.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import {DateRange} from 'react-day-picker';
+
+import {Job} from '../types/types';
+
+import {craeteJob, getJobs} from './actions';
+
+import JobDateTime from './job_date_time';
+import JobDownloadLink from './job_download_link';
+import JobScheduleModal from './job_schedule_modal';
+import './job_schedule_modal.scss';
+
+export type Props = {
+    jobs: Job[];
+    showModal: boolean;
+    className?: string;
+}
+
+type State = {
+    jobs: Job[];
+    showModal: boolean;
+    className?: string;
+}
+
+export default class JobTable extends React.Component<State, Props> {
+    static defaultProps = {
+        jobs: [],
+        showModal: false,
+    };
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {jobs: [], showModal: false};
+    }
+
+    interval: ReturnType<typeof setInterval>|null = null;
+
+    async componentDidMount() {
+        const jobs = await getJobs();
+
+        // eslint-disable-next-line react/no-did-mount-set-state
+        this.setState({jobs});
+        this.interval = setInterval(this.reload, 15000);
+    }
+
+    componentWillUnmount() {
+        if (this.interval) {
+            clearInterval(this.interval);
+        }
+    }
+
+    reload = async () => {
+        const jobs = await getJobs();
+        this.setState({jobs});
+    };
+
+    render() {
+        const createDump = async (range: DateRange | undefined) => {
+            await craeteJob(range!);
```

Comment:
This is probably not great -- if you are sure you won't have an undefined range (the `range!`), then you shouldn't have it in the function type, then you woundn't need the non-null assertion `!`